java/src/jmri/jmrit/display/logixng/WindowToFront.java

```diff
@@ -0,0 +1,301 @@
+        return CategoryDisplay.DISPLAY;
+    }
+
+    private void throwErrorJmriJFrameDoesNotExists() throws JmriException {
+        var lng = getConditionalNG();
+        var cng = getConditionalNG();
+        var m = getModule();
+        String errorMessage;
+        if (m != null) {
+            errorMessage = Bundle.getMessage(
+                    "WindowToFront_ErrorNoJmriJFrame_Module",
+                    getLongDescription(), m.getDisplayName(), getSystemName());
+        } else {
+            errorMessage = Bundle.getMessage(
+                    "WindowToFront_ErrorNoJmriJFrame_LogixNG",
+                    getLongDescription(), lng.getDisplayName(), cng.getDisplayName(), getSystemName());
+        }
+        List<String> list = Arrays.asList(errorMessage.split("\n"));
+        throw new JmriException(Bundle.getMessage("WindowToFront_ErrorNoJmriJFrame"), list);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void execute() throws JmriException {
+        JmriJFrame jmriJFrame;
+
+//        System.out.format("WindowToFront.execute: %s%n", getLongDescription());
+
+        switch (_addressing) {
+            case Direct:
+                jmriJFrame = this._jmriJFrame;
+                if (jmriJFrame == null && (_jmriJFrameTitle != null && !_jmriJFrameTitle.isBlank())) {
+                    jmriJFrame = JmriJFrame.getFrame(_jmriJFrameTitle);
+                    if (jmriJFrame == null) {
+                        log.error("ddd");
+                        throwErrorJmriJFrameDoesNotExists();
+                    }
+                }
+                break;
+
+            case Reference:
+                String ref = ReferenceUtil.getReference(
+                        getConditionalNG().getSymbolTable(), _reference);
+                jmriJFrame = JmriJFrame.getFrame(ref);
+                break;
+
+            case LocalVariable:
+                SymbolTable symbolTable = getConditionalNG().getSymbolTable();
+                jmriJFrame = JmriJFrame.getFrame(TypeConversionUtil
+                                .convertToString(symbolTable.getValue(_localVariable), false));
+                break;
+
+            case Formula:
+                jmriJFrame = _expressionNode != null ?
+                        JmriJFrame.getFrame(TypeConversionUtil
+                                        .convertToString(_expressionNode.calculate(
+                                                getConditionalNG().getSymbolTable()), false))
+                        : null;
+                break;
+
+            default:
+                throw new IllegalArgumentException("invalid _addressing state: " + _addressing.name());
+        }
+
+//        System.out.format("WindowToFront.execute: positionable: %s%n", positionable);
+
+        if (jmriJFrame == null) {
+            log.error("Frame is null");
+            return;
+        }
+
+        JmriJFrame frame = jmriJFrame;
+        ThreadingUtil.runOnLayout(() -> {
```

Comment:
Could you make this runOnGUI please?  It doesn't make a difference now, as the layout and GUI threads are the same.  But maybe someday it will make a difference.  Thanks.

Reply:
Thanks! Fixed.

Reply:
I have made this change to three other LogixNG actions that also affects the GUI. See this commit: https://github.com/JMRI/JMRI/pull/12860/commits/626a328d12c04bd4258e1a9f0a4779816deaaa02